core/src/test/java/kafka/testkit/KafkaClusterTestKit.java

```diff
@@ -174,13 +174,17 @@ private KafkaConfig createNodeConfig(TestKitNode node) {
                 props.put(KafkaConfig$.MODULE$.LogDirsProp(),
                         String.join(",", brokerNode.logDataDirectories()));
             }
-            props.put(KafkaConfig$.MODULE$.ListenerSecurityProtocolMapProp(),
-                    "EXTERNAL:PLAINTEXT,CONTROLLER:PLAINTEXT");
-            props.put(KafkaConfig$.MODULE$.ListenersProp(), listeners(node.id()));
-            props.put(KafkaConfig$.MODULE$.InterBrokerListenerNameProp(),
-                    nodes.interBrokerListenerName().value());
-            props.put(KafkaConfig$.MODULE$.ControllerListenerNamesProp(),
-                    "CONTROLLER");
+
+            // listeners could be defined via Builder::setConfigProp which shouldn't be overridden
+            if (!props.containsKey(KafkaConfig$.MODULE$.ListenersProp())) {
```

Comment:
Hm that's a good point and I did ponder over this one as well. The alternative would be to parse the user defined listener here and make sure that either the security protocol map, inter broker listener, controller listener are all set appropriately or else do so ourselves - this would involve making sure that the protocols, SASL mechanisms etc. all match up properly which seems potentially error prone (I tried a setup with broker listeners and controller listeners using different security protocols or mechanisms and I wasn't able to get it to work - not sure whether something like that is even currently supported). I'm assuming that users would likely want to set custom listeners in case they want non-plaintext listeners at which point it isn't much extra work to also setup the controller listener accordingly (and there's multiple examples on how to do so in various integration tests). What do you think?

Reply:
I think removing the `if` guard entirely and just doing `putIfAbsent` would be the most intuitive behavior. If someone still misconfigures their cluster, that's on them; we don't have to try to do any preflight checks here.

Then, if someone wants to enable SSL, they could just tweak the `listener.security.protocol.map` property and leave everything else as-is.